src/main/java/org/dcsa/tnt/service/impl/TNTEventSubscriptionTOServiceImpl.java

```diff
@@ -0,0 +1,357 @@
+            eventSubscription.setSecret(esTo.getSecret());
+            eventSubscription.setCarrierBookingReference(esTo.getCarrierBookingReference());
+            eventSubscription.setEquipmentReference(esTo.getEquipmentReference());
+            eventSubscription.setCarrierServiceCode(esTo.getCarrierServiceCode());
+            eventSubscription.setCarrierVoyageNumber(esTo.getCarrierVoyageNumber());
+            eventSubscription.setVesselIMONumber(esTo.getVesselIMONumber());
+            eventSubscription.setTransportDocumentReference(esTo.getTransportDocumentReference());
+            eventSubscription.setTransportCallID(esTo.getTransportCallID());
+            return eventSubscription;
+          };
+
+  @Override
+  public Mono<TNTEventSubscriptionTO> update(TNTEventSubscriptionTO eventSubscriptionTO) {
+    return eventSubscriptionRepository
+        .deleteEventTypesForSubscription(eventSubscriptionTO.getSubscriptionID())
+        .thenReturn(eventSubscriptionTO)
+        .map(eventSubscriptionTOToEventSubscription)
+        .flatMap(
+            updated ->
+                eventSubscriptionService
+                    .findById(updated.getSubscriptionID())
+                    .map(
+                        original -> {
+                          updated.setSecret(original.getSecret());
+                          updated.copyInternalFieldsFrom(original);
+                          return updated;
+                        }))
+        .flatMap(eventSubscriptionService::update)
+        .flatMap(ignored -> createEventTypes(eventSubscriptionTO));
+  }
+
+  private Mono<TNTEventSubscriptionTO> createEventTypes(
+      TNTEventSubscriptionTO eventSubscriptionTO) {
+
+    List<EventType> eventTypes;
+
+    if (CollectionUtils.isEmpty(eventSubscriptionTO.getEventType())) {
+      eventTypes = ALL_ALLOWED_EVENT_TYPES;
+      eventSubscriptionTO.setEventType(eventTypes);
+    } else {
+      eventTypes = eventSubscriptionTO.getEventType();
+    }
+
+    return Flux.fromIterable(eventTypes)
+        .concatMap(
+            eventType ->
+                eventSubscriptionRepository.insertEventTypeForSubscription(
+                    eventSubscriptionTO.getSubscriptionID(), eventType))
+        .then(Mono.just(eventSubscriptionTO));
+  }
+
+  private Mono<Void> createTransportDocumentEventTypes(TNTEventSubscriptionTO eventSubscriptionTO) {
+
+    if (null == eventSubscriptionTO.getTransportDocumentTypeCode()) {
+      eventSubscriptionTO.setTransportDocumentTypeCode(ALL_TRANSPORT_DOCUMENT_TYPES);
+    }
+
+    List<TransportDocumentTypeCode> transportDocumentTypeCodes =
+        eventSubscriptionTO.getTransportDocumentTypeCode();
+    return Flux.fromIterable(transportDocumentTypeCodes)
+        .flatMap(
+            td ->
+                eventSubscriptionRepository.insertTransportDocumentEventTypeForSubscription(
+                    eventSubscriptionTO.getSubscriptionID(), td))
+        .then();
+  }
```

Comment:
This method will also be needed in OVS - why not put it in EventCore? and refer to an abstract function getAllTransportDocumentTypes()